flux-middle/src/rty/projections.rs

```diff
@@ -42,13 +42,15 @@ impl<'sess, 'tcx> ProjectionTable<'sess, 'tcx> {
     fn new<T: TypeVisitable>(
         genv: &'sess GlobalEnv<'sess, 'tcx>,
         def_id: DefId,
+        param_env: &ParamEnv,
         t: &T,
     ) -> Result<Self, QueryErr> {
         let mut preds = FxHashMap::default();
 
         let mut vec = vec![];
         // 1. Insert generic predicates of the callsite `def_id`
-        vec.push(genv.predicates_of(def_id)?.skip_binder().predicates);
+        // TODO-EARLY vec.push(genv.predicates_of(def_id)?.skip_binder().predicates);
+        vec.push(param_env.predicates.clone());
```

Comment:
See See https://github.com/flux-rs/flux/pull/503#discussion_r1319146083

I believe that instead of having this extra `vec` we could have the opaque type bounds already in the param env.

This can go into the next PR that deals with dependencies in opaque types. We are skipping the binder in the bounds a few lines down, so I wonder if that early binder should be eliminated together with the binder on function predicates.

Reply:
yes, leaving this for the `impl` PR... as will be subsumed by it.